cmd/kube-scheduler/app/options/deprecated.go

```diff
@@ -120,11 +121,17 @@ func (o *DeprecatedOptions) ApplyTo(cfg *kubeschedulerconfig.KubeSchedulerConfig
 		}
 	}
 
+	// The following deprecated options affect the only existing profile that is
+	// added by default.
+	profile := &cfg.Profiles[0]
+	if len(o.SchedulerName) > 0 {
+		profile.SchedulerName = o.SchedulerName
```

Comment:
should we detect/fail if this would override a non-default profile.SchedulerName value?

Reply:
I don't follow the question.

It should affect any profile. However, this code only runs when no component config is provided.